numeric_ranges_algorithms/numeric_ranges_algorithms.md

```diff
@@ -0,0 +1,484 @@
+
+This proposal does not attempt to fill this gap in the Standard parallel algorithms, but would welcome a separate proposal to do so.  We think the right way would be to propose a new algorithm with a distinct name.  A reasonable choice of name would be `fold` (just `fold` by itself, not `fold_left` or `fold_right`).
+
+## What algorithms to include?
+
+We propose ranges overloads (both parallel and nonparallel) of only three algorithms: `reduce`, `inclusive_scan`, and `exclusive_scan`.  We also propose parallel and non-parallel convenience wrappers `ranges::sum(r)` as `ranges::reduce(r, plus{}, range_value_t<R>())` and `ranges::product(r)` as `ranges::reduce(r, multiplies{}, range_value_t<R>(1))`.
+
+### Current set of numeric algorithms
+
+<a href="https://isocpp.org/files/papers/P3179R8.html">P3179R8</a>, "C++ Parallel Range Algorithms," is in the last stages of wording review as of the publication date.  P3179R8 explicitly defers adding ranges versions of the numeric algorithms.  This proposal does that.  As such, we focus on the numeric algorithms, that is, the 11 algorithms in <a href="https://eel.is/c++draft/numeric.ops">[numeric.ops]</a>.
+
+* `iota`
+
+* `accumulate`
+
+* `inner_product`
+
+* `partial_sum`
+
+* `adjacent_difference`
+
+* `reduce`
+
+* `inclusive_scan`
+
+* `exclusive_scan`
+
+* `transform_reduce`
+
+* `transform_inclusive_scan`
+
+* `transform_exclusive_scan`
+
+We don't have to add ranges versions of all these algorithms.  Several already have a ranges version in C++23, possibly with a different name.  Some others could be omitted because they have straightforward replacements using existing views and other ranges algorithms.  We base our algorithm selection decisions on <a href="https://wg21.link/P2214R2">P2214R2</a>, "A Plan for C++23 Ranges," and <a href="https://wg21.link/P2760R1">P2760R1</a>, "A Plan for C++26 Ranges."  As P2214R2 explains, "one of the big motivations for Ranges was the ability to actually compose algorithms."  It's idiomatic for ranges to use views and projections where possible, instead of creating new algorithms.  For P2214, see in particular Section 5.  (Note that the table at the start of the section lists `shift_left` and `shift_right`.  These are not numeric algorithms, and <a href="https://isocpp.org/files/papers/P3179R8.html">P3179R8</a> adds parallel ranges versions of them.)
+
+### Algorithms that already have ranges versions
+
+The following numeric algorithms already have ranges versions in C++23.
+
+* `iota` already has a ranges version in C++23.  P3179R8 adds a parallel version.

```

Comment:
No, there is no `iota` in P3179.

We thought of adding it, as it seemed straightforward initially. But the problem is with the type requirements for the value argument of `iota`. Currently, it is `weakly_incrementable` and the only allowed operations with it are increments. That clearly is not enough for efficient parallelism, so we would need to introduce some new concept (`advanceable`?) that additionally allows at least `i + n`, ideally done in constant time. That seemed a sufficiently big and non-obvious difference to the status quo, and we decided not to go this direction just for the sake of parallel `iota`, which can also be imitated with `copy( policy, iota_view(value), result)`.

Reply:
@akukanov Thanks for explaining!  I agree that `views::iota` suffices.  Ranges' design favors composing algorithms instead of creating new ones, so I'd rather not create a `ranges::iota` algorithm.

Reply:
I've fixed this in the proposal and will update soon; thanks!